src/pages/projectSettings/tabs/EventLogTab/EventLogTab.tsx

```diff
@@ -38,11 +47,121 @@ export const EventLogTab: React.FC = () => {
+  };
+
+  const eventData = isRepo
+    ? repoEventData?.repoEvents?.eventLogEntries
+    : projectEventData?.projectEvents?.eventLogEntries;
+
+  return (
+    <div data-cy="event-log">
+      {((eventData as logEntry[]) || []).map(
+        ({ user, timestamp, before, after }) => (
+          /* @ts-expect-error */
+          <EventLogCard key={`event_log_${timestamp}`}>
+            <EventLogHeader user={user} timestamp={timestamp} />
+            <Table
+              data={getEventDiffLines(before, after)}
+              columns={[
+                <TableHeader
+                  key="key"
+                  label={<span data-cy="key">Property</span>}
+                  sortBy={(datum: EventDiffLine) => datum.key}
+                />,
+                <TableHeader
+                  key="before"
+                  label={<span data-cy="before">Before</span>}
+                  sortBy={(datum: EventDiffLine) => datum.before}
+                />,
+                <TableHeader
+                  key="after"
+                  label={<span data-cy="after">After</span>}
+                  sortBy={(datum: EventDiffLine) => datum.after}
+                />,
+              ]}
+            >
+              {({ datum }) => (
+                <Row key={datum.key} data-cy="event-log-table-row">
+                  <Cell>
+                    <StyledCell>{datum.key ?? ""}</StyledCell>
+                  </Cell>
+                  <Cell>
+                    <StyledCell>{getEventValue(datum.before)}</StyledCell>
+                  </Cell>
+                  <Cell>
+                    <StyledCell>{getEventValue(datum.after)}</StyledCell>
+                  </Cell>
+                </Row>
+              )}
+            </Table>
+          </EventLogCard>
+        )
+      )}
+    </div>
+  );
+};
+
+interface Props {
+  timestamp: Date;
+  user: string;
+}
+
+const EventLogHeader: React.VFC<Props> = ({ user, timestamp }) => (
+  <StyledHeader>
+    <H3>{getDateCopy(timestamp)}</H3>
+
+    <div> {user} </div>
+  </StyledHeader>
+);
+
+/* @ts-expect-error */
+const EventLogCard = styled(Card)`
+  margin-bottom: 48px;
+  padding: ${size.m};
+  width: 150%;
+`;
+
+const StyledCell = styled("div")`
+  word-break: break-all;
+  font-size: 12px;
+  font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
+`;
+
+const StyledHeader = styled("div")`
+  padding-bottom: ${size.l};
+  padding-left: ${size.xxs};
+`;
+
+const getEventValue = (value: any) => {
+  if (value === true) {
+    return "true";
+  }
+  if (value === false) {
+    return "false";
+  }
+  if (value === null || value === undefined) {
+    return "";
+  }
+
+  if (value === "") {
+    return '""';
+  }
+
+  const splitArray = JSON.stringify(value).split(",");
+
   return (
-    <>
-      <div>This is the event log page for {identifier}</div>
-      <div> {JSON.stringify(projectEventData)}</div>
-      <div> {JSON.stringify(repoEventData)}</div>
-    </>
+    <div>
```

Comment:
```ts
const getEventValue = (value: boolean | string): string => {
  if (value === null || value === undefined) {
    return "";
  }
  if (typeof value === "boolean") {
    return value ? "true" : "false";
  }
  const transformedValue = JSON.stringify(value);
  if (Array.isArray(transformedValue)) {
    return transformedValue.join(", ");
  }

  return transformedValue ? String(transformedValue) : "";
};

```

This outputs the same results.

Reply:
And adds some type safety.

Reply:
this solution does not handle arrays nicely
<img width="443" alt="image" src="https://user-images.githubusercontent.com/13104717/165582158-f34cb587-6a3d-49ee-8d48-cb9c211043e4.png">

Reply:
I'll take some parts of it and plug it into the existing function though

Reply:
JSON.stringify will return a string. I think you may have wanted JSON.parse

Reply:
JSON.parse doesn't work well for this use case

Reply:
```
const getEventValue = (
  value: null | undefined | boolean | string | Array<any>
) => {
  if (value === null || value === undefined) {
    return "";
  }
  if (typeof value === "boolean") {
    return String(value);
  }

  if (typeof value === "string") {
    return `"${value}"`;
  }
  if (Array.isArray(value)) {
    return JSON.stringify(value).replaceAll(",", ",\n");
  }
};
```

Produces the same output without the string parsing and extra divs
![image](https://user-images.githubusercontent.com/4605522/165842101-d5815d31-feab-4de5-8ea0-e49826afeefd.png)